staging/src/k8s.io/kubectl/pkg/polymorphichelpers/logsforobject.go

```diff
@@ -54,6 +54,7 @@ func logsForObject(restClientGetter genericclioptions.RESTClientGetter, object,
 
 // this is split for easy test-ability
 func logsForObjectWithClient(clientset corev1client.CoreV1Interface, object, options runtime.Object, timeout time.Duration, allContainers bool) (map[corev1.ObjectReference]rest.ResponseWrapper, error) {
+	options = options.DeepCopyObject()
```

Comment:
can you add unit test case for this change?

Reply:
ok.

Reply:
test case added.